batch/batch/driver/instance_collection/pool.py

```diff
@@ -159,11 +159,6 @@ def get_instance(self, user, cores_mcpu):
             if user != 'ci' or (user == 'ci' and instance.location == self._default_location()):
                 return instance
             i += 1
-        histogram = collections.defaultdict(int)
-        for instance in self.healthy_instances_by_free_cores:
-            histogram[instance.free_cores_mcpu] += 1
-        log.info(f'schedule {self}: no viable instances for {cores_mcpu}: {histogram}')
```

Comment:
The original purpose of this logging statement was to make sure the fair share algorithm was working properly and understand why we weren't scheduling for some reason. Two users could deadlock if the share they were each entitled to was greater than the amount of free cores even if theoretically both jobs could fit in what free cores was available. We fixed the deadlock by putting a random bump into the code if the deadlock was too frequent. So I think we can do without this logging statement as it's not helpful.

```python3
                if scheduled_cores_mcpu + record['cores_mcpu'] > allocated_cores_mcpu:
                    if random.random() > self.exceeded_shares_counter.rate():
                        self.exceeded_shares_counter.push(True)
                        self.scheduler_state_changed.set()
                        break
                    self.exceeded_shares_counter.push(False)
```